diff --git a/lua/lazydev/workspace.lua b/lua/lazydev/workspace.lua
index 8cb515b..81801d1 100644
--- a/lua/lazydev/workspace.lua
+++ b/lua/lazydev/workspace.lua
@@ -185,6 +185,7 @@ function M:update()
     end
   end
 
+  local defaultIgnoreDir = settings.Lua and settings.Lua.workspace and settings.Lua.workspace.ignoreDir or {}
   settings = vim.tbl_deep_extend("force", settings, {
     Lua = {
       runtime = {
@@ -195,7 +196,7 @@ function M:update()
       workspace = {
         checkThirdParty = false,
         library = library,
-        ignoreDir = Config.lua_root and { "/lua" } or nil,
+        ignoreDir = Config.lua_root and { "/lua", unpack(defaultIgnoreDir) } or defaultIgnoreDir,
       },
     },
   })

